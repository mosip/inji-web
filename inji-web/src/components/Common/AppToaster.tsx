import React from "react";
import {ToastContainer} from "react-toastify";

export const AppToaster: React.FC = () => {
    return <ToastContainer
        position="top-right"
        autoClose={1000}
        hideProgressBar
        newestOnTop
        closeOnClick
        rtl={false}
        icon={<React.Fragment/>}
        pauseOnFocusLoss
        draggable
        pauseOnHover
        style={{width: '400px'}}
        theme="colored"
    />
}
