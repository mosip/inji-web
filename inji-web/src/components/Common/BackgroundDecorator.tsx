import React from 'react';

interface BackgroundDecoratorProps {
    circleCount?: number;
    topOffset?: number;
    baseRadius?: number;
    logoSrc?: string;
    logoAlt?: string;
    testId?: string;
}

export const BackgroundDecorator: React.FC<BackgroundDecoratorProps> = ({
    circleCount = 6,
    topOffset = 155,
    baseRadius = 96,
    logoSrc,
    logoAlt = "Logo",
    testId = "background-decorator"
}) => {
    return (
        <div 
            className="overflow-hidden absolute inset-0 z-0 flex items-center justify-center pointer-events-none"
            data-testid={testId}
        >
            <div className={`absolute top-[${topOffset}px]`} style={{ top: `${topOffset}px` }}>
                {[...Array(circleCount)].map((_, index) => {
                    const radius = baseRadius + index * baseRadius;
                    const opacity = 0.8 - index * 0.1;
                    return (
                        <div
                            key={index}
                            className="absolute rounded-full border overflow-hidden"
                            style={{
                                width: `${radius}px`,
                                height: `${radius}px`,
                                borderWidth: '1px',
                                borderColor: `rgba(228, 231, 236, ${opacity})`,
                                top: `calc(50% - ${radius / 2}px)`,
                                left: `calc(50% - ${radius / 2}px)`
                            }}
                            data-testid={`${testId}-circle-${index}`}
                        />
                    );
                })}
                
                {logoSrc && (
                    <div
                        className="flex items-center justify-center"
                        data-testid={`logo-${testId}-container`}
                    >
                        <img
                            src={logoSrc}
                            alt={logoAlt}
                            data-testid={`logo-${testId}`}
                        />
                    </div>
                )}
            </div>
        </div>
    );
};