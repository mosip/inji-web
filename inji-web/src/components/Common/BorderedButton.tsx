import React from "react";

export const BorderedButton:React.FC<BorderedButtonProps> = (props) => {
  return <div className={"bg-gradient-to-r rounded-lg from-iw-primary to-iw-secondary p-0.5"}>
            <div className="py-2 px-4 rounded-lg bg-white">
                <button
                    data-testid={props.testId}
                    onClick={props.onClick}
                    className="text-iw-tertiary font-bold rounded-lg">
                    {props.title}
                </button>
            </div>
    </div>
}

export type BorderedButtonProps = {
    testId: string;
    onClick: ()=> void;
    title: string;
}
