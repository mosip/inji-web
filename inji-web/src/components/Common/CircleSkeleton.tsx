type CircleSkeletonProps = {
    size?: string;
    className?: string;
};

export const CircleSkeleton = ({
                                   size = 'w-20 h-20',
                                   className = '',
                               }: CircleSkeletonProps) => {
    return (
        <div
            className={`rounded-full bg-gray-300 animate-pulse ${size} ${className}`}
        />
    );
};
  