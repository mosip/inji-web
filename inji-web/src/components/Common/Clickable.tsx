import React from 'react';

export interface ClickableProps {
    onClick: () => void;
    children: React.ReactNode;
    testId: string;
    className: string;
}

export const Clickable: React.FC<ClickableProps> = ({
                                                        onClick,
                                                        children,
                                                        testId,
                                                        className = '',
                                                    }) => {
    const handleKeyPress = (event: React.KeyboardEvent<HTMLDivElement>) => {
        if (event.key === 'Enter' || event.key === ' ') {
            event.preventDefault();
            onClick();
        }
    };

    return (
        <div
            role="menuitem"
            tabIndex={0}
            className={className}
            onClick={onClick}
            onKeyDown={handleKeyPress}
            data-testid={testId}
        >
            {children}
        </div>
    );
};
