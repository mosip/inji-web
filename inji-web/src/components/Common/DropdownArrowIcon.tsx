import React from 'react';
import {RiArrowUpSFill, RiArrowDownSFill} from 'react-icons/ri';
import { DropdownArrowIconProps } from '../../types/data';
import { GradientWrapper } from './GradientWrapper';

const DropdownArrowIcon: React.FC<DropdownArrowIconProps> = ({isOpen, size = 20, color = 'var(--iw-color-languageArrowIcon)'}) => {
    const ArrowIcon = isOpen ? RiArrowUpSFill : RiArrowDownSFill;

    return (
        <GradientWrapper>
            <ArrowIcon size={size} color={color} />
        </GradientWrapper>
    );
};

export default DropdownArrowIcon;
