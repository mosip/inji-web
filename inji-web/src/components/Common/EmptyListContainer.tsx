import React from "react";
import {EmptyListContainerProps} from "../../types/components";

export const EmptyListContainer: React.FC<EmptyListContainerProps> = ({content}) => {
    return <React.Fragment>
        <div data-testid="EmptyList-Outer-Container"
             className="flex justify-center items-center w-full mx-auto flex-col">
            <div data-testid="EmptyList-Inner-Container" className="container mx-auto mt-8 px-4 flex-1 flex flex-col">
                <p data-testid="EmptyList-Text" className="text-center">{content}</p>
            </div>
        </div>
    </React.Fragment>
}
