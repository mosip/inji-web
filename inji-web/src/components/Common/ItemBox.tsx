import React from "react";
import {ItemBoxProps} from "../../types/components";

export const ItemBox: React.FC<ItemBoxProps> = (props) => {
    return <React.Fragment>
        <div key={props.index}
             data-testid={`ItemBox-Outer-Container-${props.index}`}
             className="bg-iw-tileBackground shadow flex flex-row shadow-iw-shadow p-4 rounded-md cursor-pointer items-center"
             onClick={props.onClick}>
            <img data-testid="ItemBox-Logo" src={props.url} alt="Issuer Logo"
                 className="w-30 h-10 justify-center me-4"/>
            <div className={"justify-center items-center"}>
                <h3 className="text-lg font-semibold text-iw-title"
                    data-testid="ItemBox-Text">{props.title}</h3>
                {props.description && <p className="text-iw-title">{props.description}</p>}
            </div>
        </div>
    </React.Fragment>
}
