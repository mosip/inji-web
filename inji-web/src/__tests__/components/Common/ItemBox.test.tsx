import React from 'react';
import {fireEvent, render, screen} from '@testing-library/react';
import {ItemBox} from "../../../components/Common/ItemBox";
describe("Test Item Box Container",() => {
    test('check the presence of the container', () => {
        const clickHandler = jest.fn();
        render(<ItemBox index={1} url={"/"} title={"TitleOfItemBox"} onClick={clickHandler} />);
        const itemBoxElement = screen.getByTestId("ItemBox-Outer-Container");
        expect(itemBoxElement).toBeInTheDocument();
    });
    test('check if content is rendered properly', () => {
        const clickHandler = jest.fn();
        render(<ItemBox index={1} url={"/"} title={"TitleOfItemBox"} onClick={clickHandler} />);
        const itemBoxElement = screen.getByTestId("ItemBox-Outer-Container");
        expect(itemBoxElement).toHaveTextContent("TitleOfItemBox")
    });

    test('check if item box onClick handler is working', () => {
        const clickHandler = jest.fn();
        render(<ItemBox index={1} url={"/"} title={"TitleOfItemBox"} onClick={clickHandler} />);
        const itemBoxElement = screen.getByTestId("ItemBox-Outer-Container");
        fireEvent.click(itemBoxElement);
        expect(clickHandler).toBeCalled()
    });
})

