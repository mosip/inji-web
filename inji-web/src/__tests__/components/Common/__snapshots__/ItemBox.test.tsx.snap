// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Layouts of ItemBox Check if the layout is matching with the snapshots 1`] = `
<DocumentFragment>
  <div
    class="bg-white w-full h-full flex flex-col shadow-iw hover:shadow-iw-hover hover:scale-105 transition-all duration-300 p-5 rounded-lg cursor-pointer items-center"
    data-testid="ItemBox-Outer-Container-1"
    role="menuitem"
    tabindex="0"
  >
    <img
      alt="Issuer Logo"
      class="w-20 h-20 object-contain mb-4"
      data-testid="ItemBox-Logo"
      src="/"
    />
    <div
      class="flex flex-col items-center"
    >
      <h3
        class="text-sm font-semibold text-gray-800 text-center w-[150px] min-h-[40px] break-words"
        data-testid="ItemBox-Text"
      >
        TitleOfItemBox
      </h3>
    </div>
  </div>
</DocumentFragment>
`;
